src/main/java/net/glease/ggfab/util/OverclockHelper.java

```diff
@@ -0,0 +1,72 @@
+package net.glease.ggfab.util;
+
+import gregtech.api.util.GT_Utility;
+
+public class OverclockHelper {
+    public static OverclockOutput normalOverclock(long recipeEUt, int duration, long inputVoltage, boolean perfectOC) {
+        if (recipeEUt > inputVoltage) return null;
+        int recipeTier = Math.max(1, GT_Utility.getTier(recipeEUt)); // ULV no overclock
+        int machineTier = GT_Utility.getTier(inputVoltage);
+        int shift = perfectOC ? 2 : 1;
+        while (recipeTier < machineTier && duration > 1) {
+            duration >>= shift;
+            recipeEUt <<= 2;
+            recipeTier++;
+        }
```

Comment:
As I general rule I think you should try and make sure you aren't accidentally Overclocking one too many times. Has happened to me previously where I have 1 UMV energy hatch, but it tries to OC to UXV and fails.

Reply:
There are unit tests for this overclock helpers. If you find a situation that it doesn't meet the expectation, please state the case clearly so I can add it to the tests and see what went wrong.

Reply:
Alright if anything like that happens I will send a message.